packages/@react-aria/interactions/src/useFocusWithin.ts

```diff
@@ -37,21 +40,21 @@ export function useFocusWithin(props: FocusWithinProps): FocusWithinResult {
     return {focusWithinProps: {}};
   }
 
-  let onFocus = createEventHandler((e: FocusEvent) => {
-    if (props.onFocusWithin) {
-      props.onFocusWithin(e);
-    }
-
+  let onFocus = (e: FocusEvent) => {
     if (!state.isFocusWithin) {
+      if (props.onFocusWithin) {
```

Comment:
This was a bug. I'm not sure why we were firing the focus within event even if focus was already within the element...